lambda-functions/elb-cleanup/event_handler.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"context"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// EventHandler the struct which will handle
+// CloudWatch events
+type EventHandler struct {
+	logger       log.FieldLogger
+	awsResourcer Resourcer
+	dryRun       bool
+}
+
+// NewEventHandler factory method to create a new
+// event handler
+func NewEventHandler(expirationDays int, awsResourcer Resourcer, dryRun bool, logger log.FieldLogger) *EventHandler {
+	return &EventHandler{
+		logger:       logger,
+		awsResourcer: awsResourcer,
+		dryRun:       dryRun,
+	}
+}
+
+// Handle the event for cloudwatch events
+func (h *EventHandler) Handle(_ context.Context, event events.CloudWatchEvent) error {
+	h.logger.Info("Unused Load Balancer(s) cleanup function called")
+
+	ctx, cancel := context.WithTimeout(context.Background(), awsTimeout)
+	defer cancel()
+
+	unUsedElbs, err := h.awsResourcer.ListUnusedElbs(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list LBs")
+	}
+	for _, lb := range unUsedElbs {
+		h.logger.Info("List of unused ALBs & NLBs are ", lb.LoadBalancerName)
+	}
+	h.logger.Info("Unused ALBs & NLBs count ", len(unUsedElbs))
+
+	if !h.dryRun {
+		// Delete unused ELBs
+		err = h.awsResourcer.DeleteElbs(ctx, unUsedElbs)
```

Comment:
you are actually doing redundant loops right?  you are already looping once here https://github.com/mattermost/mattermost-cloud-monitoring/pull/368/files#diff-632c94fa8998e81df979475b708c906096ed3afd62df4353a30d5b13fda191b4R40 you can just hook this logic of deletion and dry run there

Reply:
Done